app/src/organisms/OnDeviceDisplay/SetupNetwork/__tests__/WifiConnectionDetails.test.tsx

```diff
@@ -63,27 +67,32 @@ describe('WifiConnectionDetails', () => {
       ethernet: null,
     })
     mockUseWifiList.mockReturnValue(mockWifiList)
+    mokcNetworkDetailsModal.mockReturnValue(<div>mock NetworkDetailsModal</div>)
   })
 
   afterEach(() => {
     jest.resetAllMocks()
   })
 
   it('should render title and description', () => {
-    const [{ getByText, getByRole }] = render(props)
-    getByText('Connect via Wi-Fi')
-    getByText('Connected')
-    getByText('mockWifi')
-    getByText('IP Address: 127.0.0.100')
-    getByText('Subnet Mask: 255.255.255.230')
-    getByText('MAC Address: WI:FI:00:00:00:00')
-    getByRole('button', { name: 'Check for updates' })
+    const [{ getByText }] = render(props)
+    getByText('Wi-Fi')
+    getByText('Successfully connected to mockWifi!')
+    getByText('View network details')
+    getByText('Continue')
+  })
+
+  it('should render network details when tapping view network details', () => {
+    const [{ getByText }] = render(props)
+    const button = getByText('View network details')
+    button.click()
+    getByText('mock NetworkDetailsModal')
   })
 
   it('when clicking Check for updates button, should call mock function', () => {
-    const [{ getByRole }] = render(props)
-    const button = getByRole('button', { name: 'Check for updates' })
-    fireEvent.click(button)
+    const [{ getByText }] = render(props)
+    const button = getByText('Continue')
+    button.click()
```

Comment:
just a suggestion, you don't have to do this!

```suggestion
   getByText('Continue').click()
```

Reply:
okay, I will switch to that way.